orangecanvas/document/quickmenu.py

```diff
@@ -47,6 +46,114 @@
+                        QPointF(br.x() + lineWidth / 2, br.y() + lineWidth / 2))]
+
+        pen = QPen()
+        pen.setWidth(lineWidth)
+        pen.setColor(QColor('#e9eff2'))
+        painter.setPen(pen)
+        painter.drawLines(lines)
+
+        """ Draw icon background """
+
+        # get category color
+        brush = as_qbrush(index.data(QtWidgetRegistry.BACKGROUND_ROLE))
+        if brush is not None:
+            color = brush.color()
+        else:
+            color = QColor("FFA840")  # orange!
+
+        # (get) cache(d) pixmap
+        bg = innerGlowBackgroundPixmap(color, QSize(rect.height(), rect.height()))
+
+        # draw background
+        bgRect = QRect(tl.x(), tl.y(), rect.height(), rect.height())
+        painter.drawPixmap(bgRect, bg, bg.rect())
+
+        """ Draw icon """
+
+        # get item decoration (icon)
+        action = index.data(QtWidgetRegistry.WIDGET_ACTION_ROLE)
+        dec = action.icon()  # type: QIcon
+        decSize = option.decorationSize  # use as approximate/minimum size
+        x = rect.left() + rect.height() / 2 - decSize.width() / 2
+        y = rect.top() + rect.height() / 2 - decSize.height() / 2
+
+        # decoration rect, where the icon is drawn
+        decTl = QPointF(x, y)
+        decBr = QPointF(x + decSize.width(), y + decSize.height())
+        decRect = QRectF(decTl, decBr)
+
+        # draw icon pixmap
+        decPixmap = dec.pixmap(decSize)
+        painter.drawPixmap(decRect, decPixmap, QRectF(decPixmap.rect()))
+        # disable icon in QStyleOption
+        option.decorationSize = QSize(-1, -1)
+
+        """ Draw selected item background """
+
+        if option.state & QStyle.State_Selected:
+            backgroundRect = QRect(tl.x() + rect.height(),
+                                   tl.y(),
+                                   rect.width() - rect.height(),
+                                   rect.height())
+            grad = QLinearGradient(0, 0, 0, 1)
+            grad.setCoordinateMode(QLinearGradient.ObjectBoundingMode)
+            grad.setColorAt(0, QColor("#688EF6"))
+            grad.setColorAt(0.5, QColor("#4047F4"))
+            grad.setColorAt(1, QColor("#2D68F3"))
+            painter.fillRect(backgroundRect, grad)
+
+            # set text drawing rect
+            textRect = QRect(tl.x() + rect.height(),
+                             tl.y(),
+                             rect.width() - 2 * rect.height() + textElideOffset,
+                             rect.height())
+        else:
+            textRect = QRect(tl.x() + rect.height(),
+                             tl.y(),
+                             rect.width() - rect.height(),
+                             rect.height())
+
+        """ Draw text (configure in orange.qss) """
+
+        option.rect = textRect
+
+        # disable icon (already drawn)
+        option.decorationSize = QSize(-1, -1)
+
+        painter.restore()
+        super().paint(painter, option, index)
```

Comment:
I get a 
```
QPainter::begin: Paint device returned engine == 0, type: 3
QPainter::setCompositionMode: Painter not active
QPainter::end: Painter not active, aborted
```
on stderror when PNG backed icons (e.g. from "Orange3-Textable" add-on) are painted as selected. This call, or more correct the `option.decorationSize = QSize(-1, -1)` above seem to be the cause.

(Infrequently the application also segfaults although I am not sure this is the cause)

Reply:
I've now subclassed `QItemDelegate` as opposed to `QStyledItemDelegate`, this should work now. I've also not gotten a segfault – does the issue persist on your side?